### PR TITLE
fix: Use a different `std::thread::Thread` for `Stdin` IO

### PR DESCRIPTION
**Pull request by vrongmeal**
See: https://docs.rs/tokio/latest/tokio/io/struct.Stdin.html

> This handle is best used for non-interactive uses, such as when a
> file is piped into the application. For technical reasons, stdin is
> implemented by using an ordinary blocking read on a separate thread,
> and it is impossible to cancel that read. This can make shutdown of
> the runtime hang until the user presses enter.
>
> For interactive uses, it is recommended to spawn a thread dedicated to
> user input and use blocking IO directly in that thread.

Resolves: #61

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -344,16 +349,39 @@ pub enum RecordsOut {
 }
 
 impl RecordsIn {
-    pub async fn into_reader(&self) -> std::io::Result<Box<dyn AsyncBufRead + Send + Unpin>> {
+    pub async fn into_lines_stream(
```
Comment (infiniteregrets) [inline src/main.rs:352]: `into_reader` was ok as a name here (consistent with `RecordsOut`)

Comment (vrongmeal) [reply src/main.rs:352]: Doesn't sound right since it isn't a reader anymore. Maybe return lines stream from `RecordsOut` as well and rename it?

**Review by infiniteregrets — COMMENTED** [review]
I made a change to use `std::process::exit` as normal `return Ok(())` wouldnt work.. maybe that can changed back to normal now?

File: src/main.rs [hunk 2]
```diff
@@ -344,16 +349,39 @@ pub enum RecordsOut {
...
         match self {
-            RecordsIn::File(path) => Ok(Box::new(BufReader::new(File::open(path).await?))),
-            RecordsIn::Stdin => Ok(Box::new(BufReader::new(tokio::io::stdin()))),
+            RecordsIn::File(path) => {
+                let file = File::open(path).await?;
+                Ok(Box::pin(LinesStream::new(
+                    tokio::io::BufReader::new(file).lines(),
+                )))
+            }
+            RecordsIn::Stdin => Ok(Box::pin(stdio_lines_stream(std::io::stdin()))),
```
Comment (infiniteregrets) [inline src/main.rs:362]: lets use std for stdout as well then?

**Comment by vrongmeal** [comment]
> I made a change to use std::process::exit as normal return Ok(()) wouldnt work.. maybe that can changed back to normal now?

Oh yeah, should be. Which change is this BTW? I'll revert it.

File: src/main.rs (hunk 2 above)
Comment (vrongmeal) [reply src/main.rs:362]: We can. I don't see any issues there.

File: src/main.rs (hunk 1 above)
Comment (infiniteregrets) [reply src/main.rs:352]: yup, sg

Comment (vrongmeal) [reply src/main.rs:352]: oh wait, let's not do any of it. `Stdout` is a writer.

**Comment by infiniteregrets** [comment]
> Oh yeah, should be. Which change is this BTW? I'll revert it.

can't find it, but can you just try breaking from the loop and do `return Ok(())`. try it with the starwars example, and see if the process exits on ctrl+C

File: src/main.rs (hunk 2 above)
Comment (vrongmeal) [reply src/main.rs:362]: Taking this back. `Stdout` is a writer. Completely different.

Comment (infiniteregrets) [reply src/main.rs:362]: okay yeah makes sense

**Review by infiniteregrets — APPROVED** [review]
thanks for finding and fixing this!

**Pull request merged**